cli/src/pcluster/validators/s3_validators.py

```diff
@@ -86,7 +86,17 @@ def _validate(self, url):
                 bucket = get_bucket_name_from_s3_url(url)
                 AWSApi.instance().s3.head_bucket(bucket_name=bucket)
             except AWSClientError as e:
-                self._add_failure(str(e), FailureLevel.ERROR)
+                if e.error_code == "404":
+                    self._add_failure(f"Bucket {bucket} not found: {str(e)}", FailureLevel.ERROR)
```

Comment:
Are you sure `bucket` variable is available in the except block?

Reply:
The variable is accessible